src/nlsolvers.jl

```diff
@@ -39,22 +88,55 @@ function Base.show(s::NewtonSolver)
     end
 end
 
-function solve_nonlinear!(solver::FerriteSolver{<:NewtonSolver}, problem)
-    newtonsolver = solver.nlsolver
-    maxiter = newtonsolver.maxiter
-    tol = newtonsolver.tolerance
+@doc raw"""
+    SteepestDescent(;maxiter=10, tolerance=1.e-6)
+
+Use a steepest descent solver to solve the nonlinear 
+problem r(x) = 0, which minimizes a potential \Pi with `tolerance`
+and the maximum number of iterations `maxiter`.
+
+This method is second derivative free and is not as locally limited as a Newton-Raphson scheme.
+Thus, it is especially suited for stronlgy nonlinear behavior with potentially vanishing tangent stiffnesses.
+"""
+Base.@kwdef struct SteepestDescent{LineSearch,LinearSolver,T}
+    linsolver::LinearSolver = BackslashSolver()
+    linesearch::LineSearch = ArmijoGoldstein()
+    maxiter::Int = 200
+    tolerance::T = 1e-6
+    numiter::Vector{Int} = [zero(maxiter)]  # Last step number of iterations
+    residuals::Vector{T} = zeros(typeof(tolerance),maxiter+1)  # Last step residual history
+end
+
+getsystemmatrix(problem,solver::SteepestDescent) = LinearAlgebra.I
+
+function reset!(s::SteepestDescent)
+    fill!(s.numiter, 0)
+    fill!(s.residuals, 0)
+end
+
+function Base.show(s::SteepestDescent)
+    println("Steepest Descent")
+end
+
+function solve_nonlinear!(solver::FerriteSolver{T}, problem) where T<:Union{SteepestDescent,NewtonSolver}
+    nlsolver = solver.nlsolver
+    maxiter = nlsolver.maxiter
+    tol = nlsolver.tolerance
+    ls = nlsolver.linesearch
```

Comment:
Since we now allow different nlsolvers, I think it makes sense to use `get*` access functions here...
```suggestion
    nlsolver = solver.nlsolver
    maxiter = getmaxiter(nlsolver)
    ls = getlinesearch(nlsolver)
```